File: 2864-Maximum-Odd-Binary-Number/py/main.py
class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        sth_str = s
        len_s = len(sth_str)
        zeros = []
        ones = []

        for char in sth_str:
            if char == '1':
                ones.append('1')
            else:
                zeros.append('0')

        if len(ones) == 0:
            return '0' * len_s
        elif len(ones) == 1:
            return '0' * (len_s - 1) + '1'
        else:
            ones.pop()
            return ''.join(ones) + ''.join(zeros) + '1'

File: 2864-Maximum-Odd-Binary-Number/rs/src/main.rs
fn main() {
    println!("{:?}", maximum_odd_bunary_number("100110010"));
}

    pub fn maximum_odd_binary_number(s: String) -> String {
        let sth_str = s;
        let len_s = sth_str.len();
        let mut zeros = Vec::new();
        let mut ones = Vec::new();

        for char in sth_str.chars() {
            if char == '1' {
                ones.push('1');
            } else {
                zeros.push('0');
            }
        }

        match ones.len() {
            0 => "0".repeat(len_s),
            1 => "0".repeat(len_s - 1) + "1",
            _ => {
                ones.pop();
                ones.iter().collect::<String>() + &zeros.iter().collect::<String>() + "1"
            }
        }
    }
